### PR TITLE
Fix regression on referrer filter and filter view based on referrer

### PR DESCRIPTION
**Pull request by Kdecherf**
https://github.com/mikecao/umami/commit/7530de87d233b04d88f1fefd425c822bdf7502f4 introduced a regression: the
domain parameter in `refFilter()` contains 'https://', which bypass the
new regex; thus we readd the call to `getDomainName()`.

During my tests I also found a collision between `ref` and React's Refs, thus I renamed it to `referrer`.

Finally I added support of referrer filtering in metric tables as it was already working for the chart: now all tables update when you filter on a specified referrer.

### CODE REVIEW

File: lib/filters.js [hunk 1]
```diff
@@ -71,8 +74,8 @@ export const refFilter = (data, { domain, domainOnly, raw }) => {
 
       if (protocol.startsWith('http')) {
         const path = removeTrailingSlash(pathname);
-        const ref = searchParams.get('ref');
-        const query = ref ? `?ref=${ref}` : '';
+        const referrer = searchParams.get('ref');
```
Comment (mikecao) [inline lib/filters.js:77]: The .get is still calling `ref`, is this intended?

File: pages/api/website/[id]/metrics.js [hunk 2]
```diff
@@ -75,6 +75,7 @@ export default async (req, res) => {
         {
           domain,
           url: type !== 'url' && url,
+          referrer: referrer,
```
Comment (mikecao) [inline pages/api/website/[id]/metrics.js:78]: Can just be `referrer`.

**Review by mikecao — COMMENTED** [review]
Looks good 👍! Just a few minor issues.

File: lib/filters.js (hunk 1 above)
Comment (Kdecherf) [reply lib/filters.js:77]: Ah, good catch, will update

File: pages/api/website/[id]/metrics.js (hunk 2 above)
Comment (Kdecherf) [reply pages/api/website/[id]/metrics.js:78]: Noted, will update :+1:

**Pull request merged**